cmd/util/ledger/util/registers/registers.go

```diff
@@ -84,11 +84,53 @@ func (b *ByAccount) ForEach(f ForEachCallback) error {
 	return nil
 }
 
-func (b *ByAccount) Payloads() []*ledger.Payload {
-	payloads := make([]*ledger.Payload, 0, len(b.registers)*payloadsPerAccountEstimate)
-	for _, accountRegisters := range b.registers {
-		payloads = accountRegisters.appendPayloads(payloads)
+func (b *ByAccount) DestructIntoPayloads(nWorker int) []*ledger.Payload {
+	payloads := make([]*ledger.Payload, b.Count())
+
+	type job struct {
+		registers *AccountRegisters
+		payloads  []*ledger.Payload
+	}
+
+	var wg sync.WaitGroup
+
+	jobs := make(chan job)
```

Comment:
Maybe use a buffered channel.

Reply:
Does it make any difference?

Reply:
Switched to a buffered channel here and in the other recently added channels: [6b423c3](https://github.com/onflow/flow-go/pull/5914/commits/6b423c330f5c1fb831eba8208d8309c7ce91e81c)